### PR TITLE
Add a testcase to for L2 configuration.

### PR DESCRIPTION
**Pull request by hdwhdw**
### Description of PR

Summary:
Fixes # (issue)

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [x ] Test case(new/improvement)

### Back port request
- [ ] 202012
- [ ] 202205
- [ ] 202305
- [ ] 202311
- [ ] 202405

#### What is the motivation for this PR?
Provide test coverage for L2 configuration
#### How did you do it?
Create a testcase that perform an L2 configuration step
#### How did you verify/test it?
on VS
#### Any platform specific information?
No
#### Supported testbed topology if it's a new test case?
Any

### CODE REVIEW

File: tests/l2/test_l2_config.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+                  f" {len(lines1)} != {len(lines2)}")
+
+    for line1, line2 in zip(lines1, lines2):
+        if (not any(ignore in line1 for ignore in DB_DUMP_IGNORE) and
+                not any(ignore in line2 for ignore in DB_DUMP_IGNORE)):
+            pytest_expect(line1 == line2, "Unequal lines " +
+                          line1 + " != " + line2)
+
+
+def test_l2_configure(request, duthosts, rand_one_dut_hostname, localhost):
```
Comment (ganglyu) [inline tests/l2/test_l2_config.py:48]: Why do you need localhost?

**Comment by ganglyu** [comment]
We have other solution for l2 config, please compare with your PR.

File: tests/l2/test_l2_config.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    hwsku = duthost.facts["hwsku"]
+    db_before = duthost.shell(DUMP_DB_CMD)["stdout"]
+    create_checkpoint(duthost)
+
+    # Perform L2 configuration
+    l2_cfg = "sudo sonic-cfggen --preset l2 -p -H -k {}" \
+        " | sudo config load /dev/stdin -y".format(hwsku)
+    duthost.shell(l2_cfg)
+    duthost.command("sudo config qos reload --no-dynamic-buffer")
+    duthost.command("sudo config save -y")
```
Comment (ganglyu) [inline tests/l2/test_l2_config.py:67]: No need to save

File: tests/l2/test_l2_config.py [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    db_after = duthost.shell(DUMP_DB_CMD)["stdout"]
+    compare_db_dump(db_before, db_after)
+
+    # Cleanup and restore.
+    rollback(duthost)
```
Comment (ganglyu) [inline tests/l2/test_l2_config.py:72]: rollback is not stable, and config reload is better.

File: tests/l2/test_l2_config.py [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+pytestmark = [
+    pytest.mark.topology('any')
+]
+
+
+def compare_db_dump(db_before, db_after):
+    """
+    Compare two db dumps and after the L2 configuration steps.
+    Assert there are no additional configuration coming from minigraph.
```
Comment (ganglyu) [inline tests/l2/test_l2_config.py:31]: I don't understand this test, how do we introduce additional configuration coming from minigraph?

**Comment by hdwhdw** [comment]
Closing for pushing a wrong commit.

**Comment by hdwhdw** [comment]
Will create another PR.

**Pull request closed without merging**